homeassistant/components/sensibo/switch.py

```diff
@@ -0,0 +1,122 @@
+
+
+@dataclass
+class SensiboDeviceSwitchEntityDescription(
+    SwitchEntityDescription, DeviceBaseEntityDescriptionMixin
+):
+    """Describes Sensibo Switch entity."""
+
+
+DEVICE_SWITCH_TYPES: tuple[SensiboDeviceSwitchEntityDescription, ...] = (
+    SensiboDeviceSwitchEntityDescription(
+        key="timer_on_switch",
+        device_class=SwitchDeviceClass.SWITCH,
+        name="Timer",
+        icon="mdi:timer",
+        value_fn=lambda data: data.timer_on,
+        extra_fn=lambda data: {"id": data.timer_id, "turn_on": data.timer_state_on},
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Sensibo binary sensor platform."""
+
+    coordinator: SensiboDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    entities: list[SensiboDeviceSwitch] = []
+
+    entities.extend(
+        SensiboDeviceSwitch(coordinator, device_id, description)
+        for description in DEVICE_SWITCH_TYPES
+        for device_id, device_data in coordinator.data.parsed.items()
+        if device_data.model != "pure"
+    )
+
+    async_add_entities(entities)
+
+
+class SensiboDeviceSwitch(SensiboDeviceBaseEntity, SwitchEntity):
+    """Representation of a Sensibo Device Switch."""
+
+    entity_description: SensiboDeviceSwitchEntityDescription
+
+    def __init__(
+        self,
+        coordinator: SensiboDataUpdateCoordinator,
+        device_id: str,
+        entity_description: SensiboDeviceSwitchEntityDescription,
+    ) -> None:
+        """Initiate Sensibo Device Switch."""
+        super().__init__(
+            coordinator,
+            device_id,
+        )
+        self.entity_description = entity_description
+        self._attr_unique_id = f"{device_id}-{entity_description.key}"
+        self._attr_name = f"{self.device_data.name} {entity_description.name}"
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return True if entity is on."""
+        return self.entity_description.value_fn(self.device_data)
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the entity on."""
+        new_state = bool(self.device_data.ac_states["on"] is False)
+        params = {
+            "minutesFromNow": 60,
+            "acState": {**self.device_data.ac_states, "on": new_state},
+        }
+        result = await self.async_send_command("set_timer", params)
+
+        if result["status"] == "success":
+            return await self.coordinator.async_request_refresh()
+        raise HomeAssistantError(f"Could not enable timer for device {self.name}")
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the entity off."""
+        result = await self.async_send_command("del_timer")
+
+        if result["status"] == "success":
+            return await self.coordinator.async_request_refresh()
+        raise HomeAssistantError(f"Could not disable timer for device {self.name}")
```

Comment:
If the plan is to also do this for the `pure_boost` it might make sense to define these in the entity description